src/mapwindow.jl

```diff
@@ -41,78 +48,176 @@ and then `mapwindow(f, img, (m,n))` should filter at the 75th quantile.
+    offset = offset1 .- stride
+    _AffineTransformer(offset, stride)
 end
 
-mapwindow(f, img::AbstractArray, window::Indices; kwargs...) =
-    mapwindow(f, img, window, "replicate"; kwargs...)
-mapwindow(f, img::AbstractVector, window::AbstractUnitRange; kwargs...) =
-    mapwindow(f, img, (window,); kwargs...)
+function _IndexTransformer(ranges::NTuple{N,Base.OneTo}) where {N}
+    _IdentityTransformer()
+end
 
-function mapwindow(f, img::AbstractArray, window::Indices, border::AbstractString;
-                   kwargs...)
-    mapwindow(f, img, window, borderinstance(border); kwargs...)
+function _IndexTransformer(ranges::NTuple{N,AbstractUnitRange}) where {N}
+    offset1 = map(first, ranges)
+    offset = offset1 .- 1
+    _OffsetTransformer(offset)
+end
+
+function unitrange_same_offset(r::Range)
+    a = first(r)
+    b = a + length(r) - 1
+    ret = a:b
+    @assert length(ret) == length(r)
+    @assert first(ret) == first(r)
+    ret
+end
+unitrange_same_offset(r::AbstractUnitRange) = r
+function unitindices_same_offset(ranges)
+    map(unitrange_same_offset, ranges)
+end
+
+# Return indices of elements of `r` that are also elements of `full`.
+function _intersectionindices(full::AbstractUnitRange, r::Range)
+    r_sub = intersect(full, r)
+    if isempty(r_sub)
+        ret = 1:0
+    else
+        ret = _indexof(r,first(r_sub)):_indexof(r,last(r_sub))
+    end
+    @assert intersect(full, r) == r[ret]
+    ret
 end
-function mapwindow(f, img::AbstractVector, window::AbstractUnitRange, border::AbstractString;
-                   kwargs...)
-    mapwindow(f, img, (window,), border; kwargs...)
+
+function _indexof(r::Range, x)
+    T = eltype(r)
+    @assert x ∈ r
+    i = one(T) + T((x - first(r)) / step(r))
+    @assert r[i] == x
+    i
 end
 
-mapwindow(f, img, window::AbstractArray, args...; kwargs...) = mapwindow(f, img, (window...,), args...; kwargs...)
+function _indices_of_interiour_range(
+        fullimgr::AbstractUnitRange, 
+        imgr::Range,
+        kerr::Range)
+    kmin, kmax = extrema(kerr)
+    idx1 = _intersectionindices(fullimgr, kmin + imgr)
+    idx2 = _intersectionindices(fullimgr, kmax + imgr)
+    idx = intersect(idx1, idx2)
+    @assert imgr[idx] + kmin ⊆ fullimgr
+    @assert imgr[idx] + kmax ⊆ fullimgr
+    idx
+end
 
-function mapwindow(f,
-                   img::AbstractArray{T,N},
-                   window::Indices{N},
-                   border::BorderSpecAny;
-                   callmode=:copy!) where {T,N}
-    _mapwindow(replace_function(f), img, window, border, default_shape(f); callmode=callmode)
+function _indices_of_interiour_indices(fullimginds, imginds, kerinds)
+    map(_indices_of_interiour_range, fullimginds, imginds, kerinds)
 end
-function _mapwindow(f,
+
+# replace median by ... outside of _mapwindow_kernel
+function _mapwindow_kernel(f,
                     img::AbstractArray{T,N},
-                    window::Indices{N},
+                    window::NTuple{N,Range},
```

Comment:
How about an `AbstractUnitRange` here, since I think we wouldn't properly handle a non-1 step if it were supplied.